mvc/controllers/areas.rst

```diff
@@ -1,8 +1,93 @@
 
-.. _issue: https://github.com/aspnet/Docs/issues/121
+- Project name
+
+  - Areas
+
+    - Products
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+    - Services
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+Once you've defined the folder hierarchy, you need to tell MVC that each controller is associated with an area. You do that by decorating the controller name with the ``[Area]`` attribute.
+
+.. code-block:: c#
+  :emphasize-lines: 4
+
+  ...
+  namespace MyStore.Areas.Products.Controllers
+  {
+      [Area("Products")]
+      public class HomeController : Controller
+      {
+          // GET: /<controller>/
+          public IActionResult Index()
+          {
+              return View();
+          }
+      }
+  }
+
+The final step is to set up a route definition that works with your newly created areas. The :doc:`routing` article goes into detail about how to create route definitions, including using conventional routes versus attribute routes. In this example, we'll use a conventional route. To do so, simply open the ``startup.cs`` file and modify it by adding the highlighted route definition below.
+
+.. code-block:: c#
+  :emphasize-lines: 4-6
+
+  ...
+  app.UseMvc(routes =>
+  {
+    routes.MapRoute(name: "areaRoute",
+      template: "{area:exists}/{controller=Home}/{action=Index}");
+
+    routes.MapRoute(
+        name: "default",
+        template: "{controller=Home}/{action=Index}");
+  });
+
+Now, when the user browses to *http://<yourApp>/products*, the ``Index`` action method of the ``HomeController`` in the ``Products`` area will be invoked.
+
+Linking between areas
+---------------------
+
+To link between areas, you simply specify the area in which the controller is defined. If the controller doesn't exist within an area, use an empty string.
+
+The following snippet shows how to link from to a controller action that is defined within an area named *Products*.
+
+.. code-block:: c#
+
+  @Html.ActionLink("See Products Home Page", "Index", "Home", new { area = "Products" }, null)
+
+To link to a controller action that is outside an area, simply specify an empty string.
+
+.. code-block:: c#
+
+  @Html.ActionLink("Go to Home Page", "Index", "Home", new { area = "" }, null)
+
+Summary
+-------
+Areas are a very useful tool for grouping semantically-related controllers and actions under a common parent folder. In this article, you learned how to set up your folder hierarchy to support ``Areas``, how to specify the ``[Area]`` attribute to denote a controller as belonging to a specified area, and how to define your routes with areas. Finally, you saw how to link between areas - including linking from outside to inside, and linking from inside to outside.
```

Comment:
With this revision, you got rid of the "outside to inside" terminology in the previous section (which I approve :smiley:) so maybe delete the last phrase, "including....."